In [10]:
import pandas as pd
import numpy as np
path = '/content/drive/My Drive/pp/'

In [63]:
train = pd.read_csv(path+'train.csv')
test = pd.read_csv(path+'test_x.csv')
submission = pd.read_csv(path+'sample_submission.csv')

In [5]:
train.info()
train.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45532 entries, 0 to 45531
Data columns (total 78 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   index       45532 non-null  int64  
 1   QaA         45532 non-null  float64
 2   QaE         45532 non-null  int64  
 3   QbA         45532 non-null  float64
 4   QbE         45532 non-null  int64  
 5   QcA         45532 non-null  float64
 6   QcE         45532 non-null  int64  
 7   QdA         45532 non-null  float64
 8   QdE         45532 non-null  int64  
 9   QeA         45532 non-null  float64
 10  QeE         45532 non-null  int64  
 11  QfA         45532 non-null  float64
 12  QfE         45532 non-null  int64  
 13  QgA         45532 non-null  float64
 14  QgE         45532 non-null  int64  
 15  QhA         45532 non-null  float64
 16  QhE         45532 non-null  int64  
 17  QiA         45532 non-null  float64
 18  QiE         45532 non-null  int64  
 19  QjA         45532 non-nul

index    0
QaA      0
QaE      0
QbA      0
QbE      0
        ..
wr_09    0
wr_10    0
wr_11    0
wr_12    0
wr_13    0
Length: 78, dtype: int64

dtypes: float64(20), int64(54), object(4)

Length: 78

45532 entries, 0 to 45531
Data columns (total 78 columns)

In [7]:
train.tail()

index  QaA   QaE  QbA   QbE  ...  wr_09  wr_10  wr_11  wr_12  wr_13
45527  45527  2.0  1050  5.0   619  ...      1      1      0      1      1
45528  45528  2.0   581  3.0  1353  ...      1      1      0      1      1
45529  45529  4.0   593  1.0   857  ...      0      1      0      1      1
45530  45530  1.0   747  3.0  1331  ...      1      1      0      1      1
45531  45531  3.0   496  5.0  1827  ...      0      1      0      1      1

[5 rows x 78 columns]

In [11]:
# 문자전환
train = train.replace(' ', '')

# 만약 결측치가 문자열 스페이스(' ')로 되어 있다면, np.nan으로 바꾸어 Pandas 라이브러리가 인식할수 있도록 변환
train = train.replace('', np.nan)

# 결측 row 제거하는 방법
train.dropna(how='all') # 'all':한 행이 모두 missing value이면 제거, any': 행 내에서 하나라도

# 결측치 처리
train.fillna(0, inplace=True)

In [24]:
train.voted.value_counts()

2    24898
1    20634
Name: voted, dtype: int64

In [40]:
# 범주형 변수 처리
print(train.race.value_counts()) 
print('\n')
print(train.age_group.value_counts())
print('\n')
print(train.gender.value_counts())
print('\n')
print(train.religion.value_counts())


White                    31248
Asian                     6834
Other                     4330
Black                     2168
Native American            548
Arab                       351
Indigenous Australian       53
Name: race, dtype: int64


10s     14215
20s     14112
30s      7836
40s      5051
50s      2889
60s      1194
+70s      235
Name: age_group, dtype: int64


Male      24217
Female    21315
Name: gender, dtype: int64


Atheist                 10192
Agnostic                 9624
Christian_Catholic       6431
Christian_Other          5137
Christian_Protestant     4875
Other                    4770
Hindu                    1429
Muslim                   1192
Buddhist                  850
Jewish                    487
Christian_Mormon          428
Sikh                      117
Name: religion, dtype: int64


In [43]:
## MEAN ENCODING

target = 'voted'

gender_mean = train.groupby('gender')[target].mean()

race_mean = train.groupby('race')[target].mean()

age_group_mean = train.groupby('age_group')[target].mean()

religion_mean = train.groupby('religion')[target].mean()


In [64]:
# 기존 변수에 encoded 된 값을 매핑 
train['gender_mean'] = train['gender'].map(gender_mean) 
train['race_mean'] = train['race'].map(race_mean) 
train['age_group_mean'] = train['age_group'].map(age_group_mean) 
train['religion_mean'] = train['religion'].map(religion_mean) 


In [50]:
X = train.drop(['voted','gender','religion','age_group','race'], axis = 1)
y = train['voted']
print(X.shape)
print(y.shape)

(45532, 77)
(45532,)


In [56]:
X.head()

index  QaA   QaE  QbA  ...  gender_mean  race_mean  age_group_mean  religion_mean
0      0  3.0   363  4.0  ...     1.539573   1.495776        1.411179       1.582809
1      1  5.0   647  5.0  ...     1.539573   1.704565        1.469671       1.613016
2      2  4.0  1623  1.0  ...     1.553206   1.495776        1.411179       1.582809
3      3  3.0   504  3.0  ...     1.539573   1.704565        1.469671       1.613016
4      4  1.0   927  1.0  ...     1.553206   1.495776        1.469671       1.542602

[5 rows x 77 columns]